packages/engine/Source/Scene/PickedMetadataInfo.js

```diff
@@ -29,11 +35,30 @@ function PickedMetadataInfo(schemaId, className, propertyName, classProperty) {
    * @type {string}
    */
   this.propertyName = propertyName;
+
   /**
-   * The optional ID of the metadata schema
+   * The the `MetadataClassProperty` that is described by this
+   * structure, as obtained from the `MetadataSchema`
    *
    * @type {MetadataClassProperty}
    */
   this.classProperty = classProperty;
+
+  /**
+   * The the `PropertyTextureProperty` or `PropertyAttributeProperty`
+   * that is described by this structure, as obtained from the
+   * property texture or property attribute of the `StructuralMetadata`
+   * that matches the class name and property name.
+   *
+   * The `PropertyTextureProperty` and `PropertyAttributeProperty`
+   * types are not public and do not share a common base type.
+   * But they both define the getters
+   * - `hasValueTransform` (`boolean`)
+   * - `offset` (`any`)
+   * - `scale` (`any`)
+   *
+   * @type {object}
+   */
+  this.metadataProperty = metadataProperty;
```

Comment:
Related to https://github.com/CesiumGS/3d-tiles/issues/650 ...

If the `hasValueTransform/offset/scale` were defined in some common `TransformableValue` interface, then this would be sufficient. But as it stands, `PropertyTextureProperty` and `PropertyAttributeProperty` are distinct, unrelated classes, and it is not clear whether the consumers of the `metadataProperty` will ever try to access the `textureReader` of the former.

I _think_ that it is important to *know* what the users of this class rely on. But if desired, I can remove that information.

Reply:
I agree with the sentiment. Class-design aside, that _should_ the point of providing type information.

Reply:
So... I assume that this is resolved...? 
(If the solution is "REMOVE IT!", then you may have to write "REMOVE IT!" - I'm not soooo good at this 😆 )

Reply:
Sorry for the ambiguity: REMOVE IT! 😆

Reply:
Unresolving: 

I stumbled over this occasionally. For the suggested change and the resulting build error at https://github.com/CesiumGS/cesium/actions/runs/11584611448/job/32252079758?pr=12237#step:6:2897 , there are basically the two-and-a-half options:

- Refer to the type without including it (causing the current error)
- Include the type (which causes a linting error)
- Do not refer to the type - which defeats the purpose of the type information

@ggetz  Any pattern of how to solve this?

Reply:
I think this is occurring because `PropertyTextureProperty` are `PropertyAttributeProperty` are marked as private. (They are also both marked as experimental, and that isn't necessary if the API is private.) Otherwise, there should be no need to import the class for the typing to work.

When we take a pass on the metadata typing, we should probably check to make sure private classes are appropriately marked as such or not.

Ok to skip for now.